integrationtests/base.go

```diff
@@ -0,0 +1,139 @@
+package integrationtests
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+)
+
+// UpWg used for setting up the wireguard interface
+func UpWg(wgConfig, wgConfDir string, t *testing.T) (string, error) {
```

Comment:
no need to pass the t struct to this function, the function only needs the tempDir to create the config in which is `wgConfDir`